src/statcache.c

```diff
@@ -966,11 +966,13 @@ bool stat_cache_dir_has_child(stat_cache_t *cache, const char *path) {
     log_print(LOG_DEBUG, SECTION_STATCACHE_CACHE, "stat_cache_dir_has_child(%s)", path);
 
     iter = stat_cache_iter_init(cache, path);
-    if ((entry = stat_cache_iter_current(iter))) {
+    while ((entry = stat_cache_iter_current(iter))) {
         // Ignore negative (non-existent) entries, those tagged with st_mode == 0
         if (entry->value->st.st_mode != 0) {
             has_children = true;
             log_print(LOG_DEBUG, SECTION_STATCACHE_CACHE, "stat_cache_dir_has_child(%s); entry \'%s\'", path, entry->key);
+            free(entry);
```

Comment:
I'd prefer that we make use of a singular `free()` for all loop iterations rather than `free()` + `break` for some. Alternatively, it can work to set the pointer to a null value and only free if it's non-null at the end of the loop.